apps/ewallet/lib/ewallet/validators/transaction_consumption_validator.ex

```diff
@@ -149,6 +149,9 @@ defmodule EWallet.TransactionConsumptionValidator do
       :cancelled_transaction_request ->
         :ok
 
+      "expired_transaction_request" ->
```

Comment:
How about applying `String.to_existing_atom/1` since `|> Kernel.||(request.expiration_reason)`? 
This way we have more consistent typing + work is done closer to the source.

Reply:
I like the idea.

Reply:
The `request.expiration_reason` can be nil, so need to validate it before converting to :atom.

In my thought, it will be become this.

```elixir
expiration_reason =
      request.expiration_reason &&
        String.to_existing_atom(request.expiration_reason)

    request
    |> TransactionRequest.valid?()
    |> Kernel.||(expiration_reason)
    |> Kernel.||(TransactionRequest.get_cancelled_error(request))
    |> do_validate_transaction_request(operation)
```
Not sure if I have a better choice 🤔

Reply:
Looks fine to me, although I'd like it explicit than a boolean shortcut e.g.

```elixir
expiration_reason =
  case request.expiration_reason do
    nil -> nil
    reason -> String.to_existing_atom(reason)
  end
```

Could turn this into its own function in `TransactionRequest` as well, can imagine it being re-used in other places.

Reply:
oh right would be better to move to `TransactionRequest` 🙇